command/agent/retry_join.go

```diff
@@ -42,23 +52,56 @@ type retryJoiner struct {
 	logger *log.Logger
 }
 
+// Validate ensures that the configuration passes validity checks for the
+// retry_join stanza. If the configuration is not valid, returns an error that
+// will be displayed to the operator, otherwise nil.
+func (r *retryJoiner) Validate(config *Config) error {
+
+	// If retry_join is defined for the server, ensure that deprecated
+	// fields and the server_join stanza are not both set
+	if config.Server != nil && config.Server.ServerJoin != nil {
+		if len(config.Server.RetryJoin) != 0 {
+			return fmt.Errorf("server_join and retry_join cannot both be defined; try defining only server_join")
```

Comment:
try defining only -> prefer

Reply:
server_join and retry_join cannot both be defined; prefer setting the server_join parameter.